src/mnode/mnode-rpc.cpp

```diff
@@ -2168,36 +2176,119 @@ As json rpc
             break;
 
         case RPC_CMD_LIST::sell:
+        {
+            std::string pastelID;
+
+            if (params.size() > 2 && params[2].get_str() != "all" && params[2].get_str() != "available" && params[2].get_str() != "unavailable"
```

Comment:
we have 3 similar like blocks with difference in first condition with params[2] checking. probably write some one function?

Reply:
the comparing is different, and the inside-block code is just too short to create a new function

Reply:
It is not bug, I just mean to place all this code in one function as it is repeated in 3 times:

template <typename F>
bool ParseParams(const UniValue& params, F f, int& minheight, std::string& pastelID) {
            if (f()) {
                if (params[2].get_str().find_first_not_of("0123456789") == std::string::npos) {
                    // This means min_height is input.
                    minheight = get_number(params[2]);
                } else {
                    // This means pastelID is input
                    pastelID = params[2].get_str();
                }
            } else if (params.size() > 2) {
                filter = params[2].get_str();
                if (params.size() > 3) {
                    if (params[3].get_str().find_first_not_of("0123456789") == std::string::npos) {
                        // This means min_height is input.
                        minheight = get_number(params[3]);
                    } else {
                        // This means pastelID is input
                        pastelID = params[3].get_str();
                    }
                }
                if (params.size() > 4) {
                    pastelID = params[3].get_str();
                    minheight = get_number(params[4]);
                }
            }
}
auto func1 = [params]() -> bool {
  return params.size() > 2 && params[2].get_str() != "all" && params[2].get_str() != "available" && params[2].get_str() != "unavailable" && params[2].get_str() != "expired" && params[2].get_str() != "sold"
}
ParseParams(params, func1, minheight, pastelID)

Reply:
Advantages:
     - Save some lines of code, because it repeats 3 times, so we save 2 blocks of repeated code.
Disadvantages:
     - Extendibility: Buy, Sell, Trade tickets contains different properties. Potentially these commands will be extended in the future to support filter by those properties. The "common" part is no longer common.
     - Giving another level of complexity when reading the code.
After considering, I don't think it's worth to implement that common function.